pkg/util/helper/binding.go

```diff
@@ -111,14 +111,20 @@ func (a *Dispenser) TakeByWeight(w ClusterWeightInfoList) {
 		remain -= replicas
 	}
 	// TODO(Garrybest): take rest replicas by fraction part
-	for i := range result {
-		if remain == 0 {
-			break
+	for i, removed, resultLen := 0, 0, len(result); i < resultLen; i++ {
```

Comment:
Hi @chaunceyjiang, thanks for committing this PR. `Dispenser` is a pure module which aims at dividing replicas by specific weight list, it does not care about propagating. So I prefer to let this struct focus more about algorithm itself instead of this propagation side of things.
I guess we can add a post process [here](https://github.com/karmada-io/karmada/blob/c08d6dc145b34f4b9e768cdbce5877c1cf3c43ce/pkg/scheduler/core/generic_scheduler.go#LL193C4-L193C4). Eliminate all cluster with 0 replica from result and return it back and I think that could work. How do you think about it?

Reply:
> I guess we can add a post process [here](https://github.com/karmada-io/karmada/blob/c08d6dc145b34f4b9e768cdbce5877c1cf3c43ce/pkg/scheduler/core/generic_scheduler.go#LL193C4-L193C4). Eliminate all cluster with 0 replica from result and return it back and I think that could work. How do you think about it?

+1